### PR TITLE
Replies and v8

### LINKED ISSUE #642: Upgrade to gateway version v8
https://discord.com/developers/docs/change-log#api-and-gateway-v8

### PR DESCRIPTION
**Pull request by Joshix-1**
fixes #642 and adds replies

### CODE REVIEW

File: javacord-core/src/main/java/org/javacord/core/entity/channel/ServerChannelBuilderDelegateImpl.java [hunk 1]
```diff
@@ -88,14 +88,14 @@ protected void prepareBody(ObjectNode body) {
         for (Map.Entry<Long, Permissions> entry : overwrittenUserPermissions.entrySet()) {
             permissionOverwrites.addObject()
                     .put("id", Long.toUnsignedString(entry.getKey()))
-                    .put("type", "member")
+                    .put("type", "1")
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/entity/channel/ServerChannelBuilderDelegateImpl.java:91]: Does this work with a string?
In the `ServerChannelUpdaterDelegateImpl` you set it as an int?

File: javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java [hunk 2]
```diff
@@ -185,19 +194,31 @@ public MessageImpl(DiscordApiImpl api, TextChannel channel, JsonNode data) {
...
         }
 
-        if (data.has("activity") && !data.get("activity").isNull()) {
+        if (data.hasNonNull("activity")) {
             activity = new MessageActivityImpl(this, data.get("activity"));
         } else {
             activity = null;
         }
+
+        if (data.has("message_reference")) {
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java:211]: Better always use `hasNonNull`. You cannot trust Discord or its docs.
```suggestion
        if (data.hasNonNull("message_reference")) {
```

File: javacord-core/src/main/java/org/javacord/core/entity/channel/ServerChannelBuilderDelegateImpl.java (hunk 1 above)
Comment (Joshix-1) [reply javacord-core/src/main/java/org/javacord/core/entity/channel/ServerChannelBuilderDelegateImpl.java:91]: It works with Strings too, but I changed it to be an int.

File: javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java (hunk 2 above)
Comment (Joshix-1) [reply javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java]: Changed it. :)

**Review by Bastian — APPROVED** [review]

**Pull request merged**